src/detection/packages/packages_linux.c

```diff
@@ -65,6 +70,44 @@ static uint32_t getNumStringsImpl(const char* filename, const char* needle)
     return count;
 }
 
+static uint32_t getNumStringsImpl(const char* filename, const char* needle)
+{
+    int fd = open(filename, O_RDONLY | O_CLOEXEC);
+    if(fd < 0)
+        return 0;
+
+    struct stat fileStat;
+    if(fstat(fd, &fileStat) < 0)
+        return 0;
+
+    size_t fileSize = (size_t)fileStat.st_size;
+    if(fileStat.st_size <= 0 || SIZE_MAX - 4095 <= fileSize)
+    {
+        // file size is zero or too big for mmap
+        // falling back to the line by line implementation
+        return getNumStringsLegacyImpl(filename, needle);
```

Comment:
use fdopen?

Reply:
fdopen will take the fd's ownership, I think there should be no problem in adding comments to explain it.
Fixed.